markdown_lab/network/client.py

```
@@ -229,38 +371,52 @@ class CachedHttpClient(HttpClient):
 
         If a cache instance is not provided and caching is enabled in the configuration, a new cache is created; otherwise, caching is disabled.
+
+        Args:
+            config: MarkdownLabConfig instance. If None, uses default config.
+            cache: Optional cache instance. If None and caching enabled, creates new cache.
         """
+        config = config or get_config()
         super().__init__(config)
 
         if cache is None:
             from ..core.cache import RequestCache
```

Pylint (3.3.7)

[convention] Import outside toplevel (core.cache.RequestCache)

(C0415)